src/System.IdentityModel.Tokens.Jwt/JwtSecurityToken.cs

```diff
@@ -170,13 +142,12 @@ public JwtSecurityToken(string issuer = null, string audience = null, IEnumerabl
             if (expires.HasValue && notBefore.HasValue)
             {
                 if (notBefore >= expires)
-                {
-                    LogHelper.Throw(string.Format(CultureInfo.InvariantCulture, LogMessages.IDX10401, expires.Value, notBefore.Value), typeof(ArgumentException), EventLevel.Error);
-                }
+                    throw LogHelper.LogException<ArgumentException>(LogMessages.IDX10401, expires.Value, notBefore.Value);
             }
 
-            this.payload = new JwtPayload(issuer: issuer, audience: audience, claims: claims, notBefore: notBefore, expires: expires);
-            this.header = new JwtHeader(signingCredentials);
+            Payload = new JwtPayload(issuer, audience, claims, notBefore, expires);
+            Header = signingCredentials == null ? new JwtHeader() : new JwtHeader(signingCredentials);
```

Comment:
I am referring to this check: signingCredentials == null ? Earlier, we were calling new JwtHeader(signingCredentials) without checking if its null or not.